jmxtrans-core/src/test/java/com/googlecode/jmxtrans/SslJmxTransformerIT.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * The MIT License
+ * Copyright © 2010 JmxTrans team
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.googlecode.jmxtrans;
+
+import com.google.common.io.Files;
+import com.googlecode.jmxtrans.test.DummyApp;
+import com.googlecode.jmxtrans.test.IntegrationTest;
+import com.googlecode.jmxtrans.test.ExternalApp;
+import com.googlecode.jmxtrans.test.RequiresIO;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.concurrent.Callable;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@Category({IntegrationTest.class, RequiresIO.class})
+public class SslJmxTransformerIT {
+    private final File jmxTransAppStdOut = new File("target" + File.separator + "ssl-jmxtrans-out.log");
+    @Rule
+    public ExternalApp app = new ExternalApp(DummyApp.class).enableJmx(12346)
+            .enableSsl("localhost.jks", "localhost");
+    @Rule
+    public ExternalApp jmxTransApp = new ExternalApp(JmxTransDummyApp.class, "ssl-integration-test.json", "5")
```

Comment:
Probably something to do with [SSLContext](https://docs.oracle.com/javase/7/docs/api/javax/net/ssl/SSLContext.html#setDefault(javax.net.ssl.SSLContext)). It might get messy and your approach of forking a new JVM might make more sense...